core/src/main/scala/org/apache/spark/ui/jobs/JobPage.scala

```diff
@@ -32,7 +32,12 @@ private[ui] class JobPage(parent: JobsTab) extends WebUIPage("job") {
 
   def render(request: HttpServletRequest): Seq[Node] = {
     listener.synchronized {
-      val jobId = request.getParameter("id").toInt
+      val paramaterId = request.getParameter("id")
+      if (paramaterId.equals("")) {
```

Comment:
@srowen, yes, you are right.

But for case `&id=` it's return "". And I used this case for testing.